primefaces/src/main/resources/META-INF/resources/primefaces/validation/validation.validators.js

```diff
@@ -132,7 +132,7 @@ if (window.PrimeFaces) {
 
                 for (var file of value) {
                     if (allowtypesRegExp && (!allowtypesRegExp.test(file.type) && !allowtypesRegExp.test(file.name)))  {
-                        throw vc.getMessage(this.ALLOW_TYPES_MESSAGE_ID, file.name);
+                        throw vc.getMessage(this.ALLOW_TYPES_MESSAGE_ID, file.name, allowtypesRegExp.source);
```

Comment:
what returns .source?

Reply:
i hope the string (value) of the regexppattern.

Reply:
hope = you didnt test it? ;D
i woud use allowtypes, it's the string

Reply:
i tested it. actually, allowtypes could be used here, too.

Reply:
What is surely not working is the attribute messageTemplate. It has no effect anymore.  I tried to find the bug but i was not successful with my little knowledge in the code. @tandraschko please have a look at it this problem since you have done the recent revisions such that you should better know what might have broken this feature.

Reply:
@tandraschko i assume that the problem is caused by this new fragment in the file 2-fileupload.js you introduced in January 24:

```
 // we need to pass the real invisible input, which contains the filelist
                    var validationResult = PrimeFaces.validation.validate($this.jq, data.fileInput, update, true, true, true, true);
                    if (!validationResult.valid) {
                        for (let clientId in validationResult.messages) {
                            var msgs = validationResult.messages[clientId];
                            for (let msg of msgs) {
                                if (!msg.rendered) {
                                    $this.showMessage({
                                        summary: msg.summary,
                                        filename: file.name,
                                        filesize: file.size
                                    });
                                }
                            }
                        }
```

in this validate method the messages are already rendered and thus are not shown anymore via the showMessage function in which the replacement of the messageTemplate would have been performed.

Reply:
have you verified whether it is still working? my tests showed that adding the messageTemplate attribute, e.g.,

```
<p:fileUpload mode="advanced"
                              multiple="false"
                              messageTemplate="XXX{size} {name} YYY"
                              update="messages cropperPanel"
                              listener="#{cropUploaderBean.handleFileUpload}">
                    <p:validateFile sizeLimit="102400"
                                    allowTypes="/(\.|\/)(gif|jpeg|jpg|png)$/"/>
                </p:fileUpload>
```

 had also no effect when the file was uploaded/added successfully

![image](https://github.com/primefaces/primefaces/assets/10051373/6df736d9-7f95-4f18-9e1e-86b6951feaa9)

Reply:
as i said, feel free to create a new issue and add the reproducer there